R/summarise_country.R

```diff
@@ -0,0 +1,163 @@
+summarise_sheet_indicator_list <- function(billion) {
+  billion <- rlang::arg_match(billion)
+  indicator_order_b <- indicator_order %>%
+    dplyr::filter(ind %in% billion_ind_codes(billion)) %>%
+    dplyr::arrange(order) %>%
+    dplyr::select(sdg, short_name, medium_name, unit_raw, )
+
+  return(indicator_order_b)
+}
+
+#' Summarise Excel sheet 'data' for HPOP country summary
+#'
+#' `summarise_HPOP_cntry_data` summarises the 'data' sgeet for the HPOP country summary.
+#'
+#' @param iso3 ISO3 codes of country to export.
+#' @param ... Additional arguments passed to `calculate_hpop_contributions` and
+#' `calculate_hpop_billion`. Use if you need to provide additional parameters to
+#' those functions.
+#' @inherit calculate_hpop_contributions params
+#'
+#' @return list of data frames to be used in the 'data' sheet of HPOP country summary
+#' export
+#'
+#' @export
+summarise_HPOP_cntry_data <-
+  function(df,
+           iso3,
+           start_year = 2018,
+           end_year = 2019:2023,
+           ...) {
+    assert_mart_columns(df)
+    assert_years(start_year, end_year)
+
+    # Filter df for country, arrange indicators by order.
+    df_iso3 <- df %>%
+      dplyr::filter(iso3 == !!iso3) %>%
+      dplyr::left_join(dplyr::select(indicator_order, ind, order), by = "ind") %>%
+      dplyr::arrange(order) %>%
```

Comment:
So, I can see a problem deriving from the `left_join` above. If the `df` passed in to the function already has a column called `order`, then these would be transformed into `order.x` and `order.y` and following functions fail. I think rather than the join above, it would be better to have a simple function that can retrieve indicator order based on an ISO3 code. Then you can just add an order column in `dplyr::mutate()` and avoid the pitfalls of the column naming issue of the join.

In these cases, we should use a convention for "temporary" column names unless `order` is an actual column name in the output. If it's temporary, start the column name with `_` and I have often put `temp` in the name. `_order_temp` for instance. This is unlikely to ever conflict (and overwriting an existing column with that name likely not problematic).

However, I see `order` is dropped below. So what you could do, is if you had a `get_ind_order()` function, don't do `left_join() %>% arrange() %>% select()`. Just do:

`dplyr::arrange(get_ind_order(.data[[ind]])`

Reply:
Makes sense, I added a `get_ind_order` to utils-summary.R